src/Functions/reinterpretAs.cpp

```diff
@@ -242,9 +247,15 @@ class FunctionReinterpret : public IFunction
                     to.resize_fill(size);
 
                     static constexpr size_t copy_size = std::min(sizeof(From), sizeof(To));
+                    size_t offset_to = sizeof(To) > sizeof(From) ? sizeof(To) - sizeof(From):0;
```

Comment:
Let's move that into the `else` case, l. 257.

Reply:
Updated